oaknational/Oak-Web-Application
# Feat/lesq 1108/canonical share page [LESQ-1108]

## Description

Music year: {owa_music_year}

- Create canonical share page
- Update lesson share query to accomodate canonical pages
- Refactor helper functions to be shared
- Fix fallback icon on canonical lesson pages when subject can't be determined (eg Combined Science/Biology)

## Issue(s)

Fixes #
Share links from canonical pages with multiple conflating programme factors 404ing

## How to test

1. Go to https://deploy-preview-2880--oak-web-application.netlify.thenational.academy/teachers/lessons/cells
2. Click on Share activities with pupils
3. You should be directed to the Share page
4. Check the non-canonical pages are still working properly
5. Check optionality lesson pages are still working properly
6. Check downloads pages are still working properly

## Screenshots

How it used to look (delete if n/a):
<img width="600" alt="Screenshot 2024-10-14 at 10 08 47" src="https://github.com/user-attachments/assets/a5c00aed-2122-4aa2-aea4-6a52580f0359">
<img width="600" alt="Screenshot 2024-10-14 at 10 08 59" src="https://github.com/user-attachments/assets/60660b32-09c9-4d86-9612-662f751e274c">


How it should now look:
{screenshots}
<img width="600" alt="Screenshot 2024-10-14 at 10 08 38" src="https://github.com/user-attachments/assets/3735d469-87d1-45c3-8647-db33f658cd9d">
<img width="600" alt="Screenshot 2024-10-14 at 10 09 09" src="https://github.com/user-attachments/assets/9d2d20f8-13bd-41ac-b166-c022e5873511">

## Checklist

- [ ] Lesson share links for KS4 science and maths subjects direct you to the canonical share page
- [ ] Non-canonical share pages still work
- [ ] Fallback icon is 'books' and not 'error' on canonical pages
